oasis-core.git ftbfs due to a broken pinned dependency

##### SUMMARY
The `moar_transports` branch from [darosior/rust-jsonrpc](https://github.com/darosior/rust-jsonrpc) has been removed, but to build depends on it via: https://github.com/oasisprotocol/oasis-core/blob/b099666e9c5d65ce086fd12a5846ea523a6554f0/Cargo.lock#L1162

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
all

##### OASIS NODE VERSION
Confirmed on:
* b099666e9c5d65ce086fd12a5846ea523a6554f0 
* v21.3.7
* v21.3.8

##### STEPS TO REPRODUCE
Follow the build instructions and run `make` or `cargo update` within the checkout.

##### ACTUAL RESULTS
The build failed due to the branch and commit referenced being removed.

```
*** Building Rust tools...
  Installing oasis-core-tools v0.0.0 (/build/oasis/tools)
    Updating git repository `https://github.com/informalsystems/tendermint-rs`
    Updating git repository `https://github.com/darosior/rust-jsonrpc`
error: failed to get `jsonrpc` as a dependency of package `oasis-core-runtime v0.0.0 (/build/oasis/runtime)`

Caused by:
  failed to load source for dependency `jsonrpc`

Caused by:
  Unable to update https://github.com/darosior/rust-jsonrpc?branch=moar_transports#ddf40d02

Caused by:
  object not found - no match for id (ddf40d0291602dd07f04199c573df03a7b57b525); class=Odb (9); code=NotFound (-3)
make: *** [Makefile:16: build-tools] Error 101 
```

I confirmed the `moar_transports` ref is missing via:
```
$ git ls-remote https://github.com/darosior/rust-jsonrpc.git | grep moar_
00eb4ed89e60f7b761a1f20dcd464d9481d93eed        refs/heads/moar_transports_b
```

The ddf40d0291602dd07f04199c573df03a7b57b525 ref is missing entirely from that repo.

##### EXPECTED RESULTS

The build would proceed correctly.

--- a/.changelog/4439.bugfix.md
+++ b/.changelog/4439.bugfix.md
@@ -1,0 +1,1 @@
+deps: Switch to the upstream jsonrpc import

--- a/runtime/Cargo.toml
+++ b/runtime/Cargo.toml
@@ -63,8 +63,7 @@ features = ["full"]
 
 [dev-dependencies]
 # For storage interoperability tests only.
-# Temporary import till https://github.com/apoelstra/rust-jsonrpc/pull/42 is merged.
-jsonrpc = { git = "https://github.com/darosior/rust-jsonrpc", branch = "moar_transports", features = ["simple_uds"] }
+jsonrpc = { version = "0.12.1", features = ["simple_uds"]}
 tempfile = "3.2.0"
 
 [[bin]]

